### PR TITLE
Fix #3007 (Change animation speed Entity, without starting it from the beginning)

### LINKED ISSUE #3007: Change animation speed of Entity, without starting it from the beginning
There is a problem: need change the speed of Entity wheel during its rotation. I made an animated wheel in Blender and sketched a similar code:

``` lua
function vehicleWheel.on_step(self)
    self.object:set_animation({x = 1, y = 100}, rotateSpeed, 0, true)
end
```

However, I only got convulsive trembling wheel, because the function set_animation animation starts from the beginning. If I put game on pause, I see that the wheel starts to spin normally. But to set the animation speed Entity, bypassing set_animation function is not possible. Can implement animation using the functions set_bone_position, but the result is predictable. Tell me, can I have missed something, or Lua Mod API is now really does not give to do it. If not, in the future it will be added to the API? I could add a function like set_animation_speed, but not Know how it will be compatible with future changes.

### PR DESCRIPTION
**Pull request by NaruTrey**
Lua API: New argument **save_playback_pos**  in method `set_animation`, which defines policy of maintaining current position of animation

### CODE REVIEW

File: src/content_cao.cpp [hunk 1]
```diff
@@ -1492,7 +1492,14 @@ void GenericCAO::updateAnimation()
 
 	if (m_animated_meshnode->getStartFrame() != m_animation_range.X ||
 		m_animated_meshnode->getEndFrame() != m_animation_range.Y)
+                {
```
Comment (SmallJoker) [inline src/content_cao.cpp]: Please use tabs to intend. Also move the bracket up, so it looks like `m_animation_range.Y) {`

File: src/content_cao.cpp [hunk 2]
```diff
@@ -1492,7 +1492,14 @@ void GenericCAO::updateAnimation()
...
+                        f32 current_frame = m_animated_meshnode->getFrameNr();
 			m_animated_meshnode->setFrameLoop(m_animation_range.X, m_animation_range.Y);
+                        // setFrameLoop method has a side effect, it resets the current frame of the animation
+                        if (current_frame > m_animated_meshnode->getStartFrame() &&
+                                current_frame < m_animated_meshnode->getEndFrame())
+                                        m_animated_meshnode->setCurrentFrame(current_frame);
```
Comment (SmallJoker) [inline src/content_cao.cpp]: Intend here the other way -> Two tabs for the new `if` line, one for the function call.
http://dev.minetest.net/Code_style_guidelines#Spaces (see `for` loop example)

**Comment by NaruTrey** [comment]
It turns out that all formatting of a file does not meet guideline?

**Comment by SmallJoker** [comment]
No, the old code parts will be updated when someone does a rewrite in that area. Doing that in this commit is not required.
For example, I don't see any space indents in this file but the code style is mixed sometimes.

**Comment by sofar** [comment]
Wouldn't `reset = false|true` be just as easy to understand and simpler to type, and have it default to `true`?

I'm thinking `save_playback_pos` is a bit long and may be confusing

**Comment by NaruTrey** [comment]
Then, not everyone will understand what will make the argument reset, without reading further documentation. It is not obvious that reset will occur.

**Comment by sofar** [comment]
I'm not aware of the current behavior, but I'd want to have the default be consistent before/after this patch.

So if the old code resets the animation, then it needs to reset it afterwards if the parameter is omitted. And if the old code doesn't reset the animation, the new code needs to do that as well by default.

The problem I have with `save_playback_pos` is that it isn't what happens: There is no field anywhere that `playback_pos` is saved _for later use_(it is used immediately instead, and never stored in a common sense of the word, nor available for retrieval). Also, `pos` suggests a `world position`, and not an `animation frame number`. All in all `save_playback_pos` is just a bad name for the method I think.

`restart` or `reset` are nice and short, and we have (and must have) documentation to clarify this stuff. 

You can't just say "but then people need to read the documentations", because, that's already the case. They already need to read the `lua_api.txt` to make any sense of the API, and minetest's API isn't different than any other API.

This patch is missing the proper explanation of `save_playback_pos` in any case.

**Comment by sofar** [comment]
I'd still like to see the `restart` parameter explicitly documented (one line should suffice). Otherwise the code looks OK from a cursory glance.

**Comment by NaruTrey** [comment]
My English is weak, please trace the readability of documentation.

**Comment by NaruTrey** [comment]
I did not like a lot of commits, which could have been avoided, I will do a new PR, without my mistakes as a newbie to Git.

**Comment by sofar** [comment]
Don't close this. Just squash and force push to the same branch. You don't want to lose the entire discussion thread associated with the change.

**Comment by SmallJoker** [comment]
@sofar is completely right here. We don't end up here like in the minetest_game repository.
minetest/minetest_game#965
minetest/minetest_game#915
minetest/minetest_game#914
minetest/minetest_game#912
minetest/minetest_game#896
minetest/minetest_game#740

However, if you're near to giving up - simply do it like tenplus1 (^): [Share your fork](https://github.com/SmallJoker/minetest/settings/collaboration), so I can squash this easily.

**Comment by NaruTrey** [comment]
I screwed up so that fork is no longer designated for this PR, so that its opening is now impossible.

**Comment by sofar** [comment]
Please, the next time you end up in a situation like this, *ping me* so I can help you recover your tree.

Also, even if you've messed up, even now your tree can be brought back to the state you had it in before.

The last commit on your local branch of this PR is 52266f3. You can always, on your local branch, `git checkout 52266f3` and re-push to the right pr branch again.

**Pull request closed without merging**